test/libdevcrypto/LibSnark.cpp

```diff
@@ -0,0 +1,470 @@
+	primary_input.emplace_back("622860516154313070522697309645122400675542217310916019527100517240519630053");
+	primary_input.emplace_back("11094488463398718754251685950409355128550342438297986977413505294941943071569");
+	primary_input.emplace_back("6627643779954497813586310325594578844876646808666478625705401786271515864467");
+	primary_input.emplace_back("2957286918163151606545409668133310005545945782087581890025685458369200827463");
+	primary_input.emplace_back("1384290496819542862903939282897996566903332587607290986044945365745128311081");
+	primary_input.emplace_back("5613571677741714971687805233468747950848449704454346829971683826953541367271");
+	primary_input.emplace_back("9643208548031422463313148630985736896287522941726746581856185889848792022807");
+	primary_input.emplace_back("18066496933330839731877828156604");
+
+	proof.g_A.g = G1(
+		Fq("12873740738727497448187997291915224677121726020054032516825496230827252793177"),
+		Fq("21804419174137094775122804775419507726154084057848719988004616848382402162497"),
+		Fq::one()
+	);
+	proof.g_A.h = G1(
+		Fq("7742452358972543465462254569134860944739929848367563713587808717088650354556"),
+		Fq("7324522103398787664095385319014038380128814213034709026832529060148225837366"),
+		Fq::one()
+	);
+	proof.g_B.g = G2(
+		Fq2(
+			Fq("15588556568726919713003060429893850972163943674590384915350025440408631945055"),
+			Fq("8176651290984905087450403379100573157708110416512446269839297438960217797614")
+		),
+		Fq2(
+			Fq("4265071979090628150845437155927259896060451682253086069461962693761322642015"),
+			Fq("15347511022514187557142999444367533883366476794364262773195059233657571533367")
+		),
+		Fq2::one()
+	);
+	proof.g_B.h = G1(
+		Fq("2979746655438963305714517285593753729335852012083057917022078236006592638393"),
+		Fq("6470627481646078059765266161088786576504622012540639992486470834383274712950"),
+		Fq::one()
+	);
+	proof.g_C.g = G1(
+		Fq("6851077925310461602867742977619883934042581405263014789956638244065803308498"),
+		Fq("10336382210592135525880811046708757754106524561907815205241508542912494488506"),
+		Fq::one()
+	);
+	proof.g_C.h = G1(
+		Fq("12491625890066296859584468664467427202390981822868257437245835716136010795448"),
+		Fq("13818492518017455361318553880921248537817650587494176379915981090396574171686"),
+		Fq::one()
+	);
+	proof.g_H = G1(
+		Fq("12091046215835229523641173286701717671667447745509192321596954139357866668225"),
+		Fq("14446807589950902476683545679847436767890904443411534435294953056557941441758"),
+		Fq::one()
+	);
+	proof.g_K = G1(
+		Fq("21341087976609916409401737322664290631992568431163400450267978471171152600502"),
+		Fq("2942165230690572858696920423896381470344658299915828986338281196715687693170"),
+		Fq::one()
+	);
+
+
+	cout << "Verifying..." << endl;
+	bool verified = libsnark::r1cs_ppzksnark_verifier_strong_IC(_vk, primary_input, proof);
+	cout << "Verified: " << verified << endl;
+}
```

Comment:
Why is this commented out?  Does it make sense to turn this into a BOOST test case?